storage/spanstore/memory/memory.go

```diff
@@ -51,8 +51,52 @@ func NewStore() *Store {
 
 // GetDependencies returns dependencies between services
 func (m *Store) GetDependencies(endTs time.Time, lookback time.Duration) ([]model.DependencyLink, error) {
-	// TODO - go through all traces and build dependencies between start and end time
-	return []model.DependencyLink{}, nil
+	deps := map[string]*model.DependencyLink{}
+	m.Lock()
+	defer m.Unlock()
+	startTs := endTs.Add(-1 * lookback)
+	for _, trace := range m.traces {
+		if m.traceIsBetweenStartAndEnd(startTs, endTs, trace) {
+			for _, s := range trace.Spans {
+				parentSpan := m.findSpan(trace, s.ParentSpanID)
```

Comment:
remember that in the storage we can have spans with identical span/parent IDs, especially with zipkin-like tracers like jaeger-client-go. I would suggest running each trace through ID deduper first